spark/src/main/resources/error/delta-error-classes.json

```
@@ -1127,7 +1127,7 @@
   },
   "DELTA_INVALID_PROTOCOL_VERSION" : {
     "message" : [
-      "Delta protocol version is not supported by this version of Delta Lake: table requires <required>, client supports <supported>. Please upgrade to a newer release."
+      "Delta protocol version is not supported by this version of Delta Lake: table \"<tableNameOrPath>\" requires reader version <readerRequired> and writer version <writerRequired>, client supports reader versions <supportedReaders> and writer versions <supportedWriters>. Please upgrade to a newer release."
```

> I'm thinking about adding library version in the error message, what you think?
> 
> `Delta protocol version is not supported by this version of Delta Lake "3.0.0": table...`
> 
> Using `io.delta.VERSION`

Interesting idea! It does seem very relevant. Maybe can rework the message accordingly?
```suggestion
      "Unsupported Delta protocol version: table \"<tableNameOrPath>\" requires reader version <readerRequired> and writer version <writerRequired>, but Delta Lake \"<deltaVersion>\" supports reader versions <supportedReaders> and writer versions <supportedWriters>. Please upgrade to a newer release."
```